### PR TITLE
Add PR and issue templates

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description
Fix https://github.com/open-vsx/publish-extensions/issues/474

## Progress

- [x] Add issue templates
- [x] Add external issue template for requests
- [x] Add a PR template
- [x] Add a guide for publishing via GitHub CI
- [x] Add docs about problems with extensions (publishing/bugs within exts)
- [ ] Before merging ⚠️: change all the links to reflect the master (HEAD) branch

### CODE REVIEW

File: docs/external_contribution_request.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+Publish `EXTENSION_NAME` to Open VSX
+
+## Issue body
+
+Please consider publishing the extension to the OpenVSX registry.
+
+We are running the Web version of VS Code at Gitpod. The problem is, that only Microsoft products like VS Code may use the MS VS Code extensions Marketplace, so we must use Open VSX instead. It is a vendor neutral alternative. Also, products like VSCodium, Theia, code-server many other projects are using the registry to allow their users to install extensions. 
+
+The docs for publishing the extension are here: https://github.com/eclipse/openvsx/wiki/Publishing-Extensions. Technically this should be straightforward. You basically must get a token and call `ovsx publish` with it. For example, this is how RedHat publishes their java extension: redhat-developer/vscode-java@ff51dbf/Jenkinsfile#L73-L82
```
Comment (loujaybee) [inline docs/external_contribution_request.md:11]: This PR: https://github.com/open-vsx/publish-extensions/pull/483/files and this PR are closely related. 

I think we should reference the other PR docs from within this issue template (we can do in a subsequent PR, we don't need to couple them together). But it's so that we make it super clear in these docs to the extension maintainers _why_ it's important that they should publish the extensions themselves.

Comment (filiptronicek) [reply docs/external_contribution_request.md:13]: Added a mention in [`87eee1b` (#479)](https://github.com/open-vsx/publish-extensions/pull/479/commits/87eee1b075673cc149db2ea33e0def14957827a4).

**Review by Siddhant-K-code — APPROVED** [review]
LGTM 🎉

File: docs/external_contribution_request.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+## Issue body
+
+Please consider publishing the extension to the OpenVSX registry. 
+
+We are running the Web version of VS Code at Gitpod. The problem is, that only Microsoft products like VS Code may use the MS VS Code extensions Marketplace, so we must use Open VSX instead. It is a vendor neutral alternative. Also, products like VSCodium, Theia, code-server many other projects are using the registry to allow their users to install extensions. 
+
+Here's a couple of words about why it might not be the best idea to let someone from the community add this extension to [open-vsx/publish-extensions](https://github.com/open-vsx/publish-extensions): https://github.com/open-vsx/publish-extensions#when-to-add-an-extension.
+
+The docs for publishing the extension are here: https://github.com/eclipse/openvsx/wiki/Publishing-Extensions. Technically this should be straightforward. You basically must get a token and call `ovsx publish` with it. For example, this is how RedHat publishes their java extension: redhat-developer/vscode-java@ff51dbf/Jenkinsfile#L73-L82
```
Comment (akosyakov) [inline docs/external_contribution_request.md:13]:
```
The docs for publishing the extension are [here](https://github.com/eclipse/openvsx/wiki/Publishing-Extensions). Technically this should be straightforward. You get a token and call `ovsx publish` with it. For example, this is how RedHat publishes their java extension: [redhat-developer/vscode-java@ff51dbf/Jenkinsfile#L73-L82](https://github.com/redhat-developer/vscode-java/blob/5d4f8d58b8e919534800ca7bc41e8513c288f573/Jenkinsfile#L78-L82)

[Here](https://github.com/PeterWone/vsc-print/pull/121) is also an example PR to contribute release GH action which can publish to GitHub, MS and OpenVSX at the same time.
```

Comment (akosyakov) [reply docs/external_contribution_request.md:13]: It is important to use permalinks.

File: docs/external_contribution_request.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+## Issue title
+
+Publish `EXTENSION_NAME` to Open VSX
+
+## Issue body
+
+Please consider publishing the extension to the OpenVSX registry.
```
Comment (akosyakov) [inline docs/external_contribution_request.md:7]: Unfortunately MS prohibits usages of MS marketplace by any other products or redistribution vsix files from it. Because of it we kindly ask you to take the ownership of your namespace in [OpenVSX](https://open-vsx.org/) and publish there as well.

OpenVSX is a vendor neutral alternative to MS marketplace used by all other derivatives of VS Code like [VSCodium](https://vscodium.com/), [Gitpod](gitpod.io), [OpenVSCode](https://github.com/gitpod-io/openvscode-server), [Theia](https://theia-ide.org/) based IDEs, and so on.

The docs for publishing the extension are [here](https://github.com/eclipse/openvsx/wiki/Publishing-Extensions). Technically this should be straightforward. You get a token and call `ovsx publish` with it. For example, this is how RedHat publishes their java extension: [redhat-developer/vscode-java@ff51dbf/Jenkinsfile#L73-L82](https://github.com/redhat-developer/vscode-java/blob/5d4f8d58b8e919534800ca7bc41e8513c288f573/Jenkinsfile#L78-L82)

[Here](https://github.com/PeterWone/vsc-print/pull/121) is also an example PR to contribute release GH action which can publish to GitHub, MS and OpenVSX at the same time. Let @filiptronicek @akosyakov know if you are opened to a PR contributing such GH action. They would be glad to help.

Gitpod as well support open source developers by providing unlimited access to platform. Feel free to reach out via https://www.gitpod.io/docs/professional-open-source

Comment (akosyakov) [reply docs/external_contribution_request.md:7]: Let's try such template instead, please use permalinks as in text.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+## What type of PR is this?
+
+- [ ] ✨ Adding a new extension
+- [ ] 📝 Updating the scripts and/or publishing workflows
+
+## Can't manually publish
+
+For the sake of efficiency and simplicity, the easiest way to publish an extension is by having it published by its official maintainers, for more info about this please refer to the [README](https://github.com/open-vsx/publish-extensions#when-to-add-an-extension). If you are the maintainer of this very extension, instead of creating a PR, publish the extension according to the [Publishing Extensions](https://github.com/eclipse/openvsx/wiki/Publishing-Extensions) guide in our project's wiki.
+
+- [ ] I have tried reaching out to the extension maintainers about publishing this extension to OpenVSX
```
Comment (akosyakov) [inline .github/PULL_REQUEST_TEMPLATE.md:17]: We should point here to template which they should use.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+<!--
+  For Work In Progress Pull Requests, please use the Draft PR feature,
+  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.
+  
+  If you're publishing a new extension to OpenVSX, please make sure the extension has an OSI-approved open source license. If you want to have an extension with a proprietary or non-approved license, please ask its maintainers to publish it.
+-->
+
+## What type of PR is this?
```
Comment (akosyakov) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: I don't think it is important. I would like to bring attention both on issues and PR for new or broken extensions to the fact that extension should be removed from this repo and published directly to OpenVSX by the author.

File: .github/ISSUE_TEMPLATES/bug.yml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+name: Bug Report
```
Comment (akosyakov) [inline .github/ISSUE_TEMPLATES/bug.yml:1]: I think issue template should also tell go to the extension author and let's remove it from here. The bug itself means that we are not capable to maintain it well.

**Comment by akosyakov** [comment]
@filiptronicek Could you have a look at remaining comments? 🙏

File: docs/external_contribution_request.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+The docs for publishing the extension are [here](https://github.com/eclipse/openvsx/wiki/Publishing-Extensions). Technically this should be straightforward. You get a token and call `ovsx publish` with it. For example, this is how RedHat publishes their Java extension: [redhat-developer/vscode-java@ff51dbf/Jenkinsfile#L73-L82](https://github.com/redhat-developer/vscode-java/blob/5d4f8d58b8e919534800ca7bc41e8513c288f573/Jenkinsfile#L78-L82)
+
+[Here](https://github.com/PeterWone/vsc-print/pull/121) is also an example PR that contributes a release GitHub Action which can publish to GitHub, MS and OpenVSX at the same time. Let @filiptronicek or @akosyakov know if you are open to a PR contributing such GitHub Action. They would gladly help out, even if there are just some questions or suggestions.
+
+Gitpod as well supports open-source developers by providing unlimited access to their platform. Feel free to reach out via https://www.gitpod.io/docs/professional-open-source
```
Comment (filiptronicek) [inline docs/external_contribution_request.md:15]: We could make this just the raw markdown (just a md code block) for easier copying.

**Comment by akosyakov** [comment]
@filiptronicek lgtm, don't forget to clean up the history before merging

Should not we also have a template for add new extension? Which again urge don't add it here? I think otherwise users will use another template which does not bring this problem?

**Comment by filiptronicek** [comment]
I have added the issue template, anything I should add, @akosyakov?

image

File: docs/external_contribution_request.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
+## Issue title
+
+Publish `EXTENSION_NAME` to Open VSX
+
+## Issue body
+
+```md
```
Comment (loujaybee) [inline docs/external_contribution_request.md:7]: Any reason for wrapping in a code block here? As you don't get the markdown formatting?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+-->
+<!--
+
+PLEASE READ THE FOLLOWING BEFORE SUBMITTING THIS PULL REQUEST:
+
+## A note about PRs adding or fixing extensions
+
+### For extension authors
+
+Although this repository is all about extensions, it is very hard to manage houndreds of extensions just by our team, we don't have the capacity to take care of every new broken extension, which includes new extensions. If you are the author of this contributed extension, please take the time to integrate OpenVSX publishing into your extension. You can refer to our [direct publish setup](docs/direct_publish_setup.md) doc for a guide on how to do it easily.
```
Comment (loujaybee) [inline .github/PULL_REQUEST_TEMPLATE.md:15]: I'd suggest we directly copy/paste the reasoning from the README for consistency, but also as it's more accurate, and also explains a bit more about _why_ it's better for authors to publish (and not just that it's a pain for us... which is true, that also!). But I'd lead with the benefits to the extension authors and/or the problems which occur when authors don't publish themselves. 

```suggestion
`publish-extensions` exists to seed the Open VSX marketplace, and also serves as a place for extensions that cannot feasibly be published directly by the extensions authors. In the long-run it is better for extension owners to publish their own plugins because:

1. Any future issues (features/bugs) with any published extensions in Open VSX will be directed to their original repo/source-control, and not confused with this repo publish-extensions.
2. Extensions published by official authors are shown within the Open VSX marketplace as such. Whereas extensions published via publish-extensions display a warning that the publisher (this repository) is not the official author.
3. Extension owners who publish their own extensions get greater flexibility on the publishing/release process, therefore ensure more accuracy/stability. For instance, in some cases publish-extensions has build steps within this repository, which can cause some uploaded plugin versions to break (e.g. if a plugin build step changes).

If you are the author of the extension being raised in this PR, please first consider directly publishing the extension yourself. You can refer to our [direct publish setup](docs/direct_publish_setup.md) doc for a guide on how to publish your plugin to Open VSX.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  For Work In Progress Pull Requests, please use the Draft PR feature,
+  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.
+  
+  If you're publishing a new extension to OpenVSX, please make sure the extension has an OSI-approved open source license. If you want to have an extension with a proprietary or non-approved license, please ask its maintainers to publish it.
+-->
+<!--
+
+PLEASE READ THE FOLLOWING BEFORE SUBMITTING THIS PULL REQUEST:
+
+## A note about PRs adding or fixing extensions
```
Comment (loujaybee) [inline .github/PULL_REQUEST_TEMPLATE.md:11]: I'd suggest we just remove this to take out some visual noise. 

```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
+<!--
+  For Work In Progress Pull Requests, please use the Draft PR feature,
```
Comment (loujaybee) [inline .github/PULL_REQUEST_TEMPLATE.md:2]: Could we move this point below some of the others, as it's not the main point/message we're trying to send?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.
+  
+  If you're publishing a new extension to OpenVSX, please make sure the extension has an OSI-approved open source license. If you want to have an extension with a proprietary or non-approved license, please ask its maintainers to publish it.
+-->
+<!--
```
Comment (loujaybee) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: Could we remove these comments? 

```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+For the sake of efficiency and simplicity, the easiest way to publish an extension is by having it published by its maintainers, for more info about this please refer to the [README](https://github.com/open-vsx/publish-extensions#when-to-add-an-extension). If the authors are open to publish the extension to OpenVSX, you can help them by contributing a GitHub Action using our handy-dandy [direct publish setup](docs/direct_publish_setup.md) doc. 
+
+ - If the extension is unmaintained, please create an issue for it instead.
+
+-->
+
+- [ ] I have read the note above about PRs contributing or fixing extensions
+- [ ] I have tried reaching out to the extension maintainers about publishing this extension to OpenVSX 
+
+    - if you haven't, please create an issue in the extension's repo using [this template](docs/external_contribution_request.md).
```
Comment (loujaybee) [inline .github/PULL_REQUEST_TEMPLATE.md:28]: Could we bring together this line and the checkbox? 

```suggestion
- [ ] I have tried reaching out to the extension maintainers about publishing this extension to OpenVSX (if no,  please create an issue in the extension's repo using [this template](docs/external_contribution_request.md).
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+
+## Description
+
+<!-- Please do not leave this blank -->
+
+This PR does *this* and *this*.
```
Comment (loujaybee) [inline .github/PULL_REQUEST_TEMPLATE.md:34]:
```suggestion
This PR introduces X change for Y reason. 
```

File: docs/external_contribution_request.md (hunk 8 above)
Comment (filiptronicek) [reply docs/external_contribution_request.md:7]: @loujaybee we decided it would be better for peeps who just go here to copy the template, so they don't copy just the rendered text.

File: docs/direct_publish_setup.md [hunk 15]
```diff
@@ -0,0 +1,26 @@
+# Publishing an extension to OpenVSX on your own
+
+It is hightly advised that extension authors publish their extensions to OpenVSX as a part of their CI/CD process. See [our reasoning why that is](https://github.com/open-vsx/publish-extensions#when-to-add-an-extension).
```
Comment (loujaybee) [inline docs/direct_publish_setup.md:3]:
```suggestion
We advise extension authors to publish their extensions to OpenVSX as a part of their CI/CD process. See [our reasoning why that is](https://github.com/open-vsx/publish-extensions#when-to-add-an-extension).
```

File: docs/direct_publish_setup.md [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+
+We have a simple template for setting a workflow with GitHub Actions which can help you kickstart your own. This template includes some neat features like:
```
Comment (loujaybee) [inline docs/direct_publish_setup.md:5]:
```suggestion
To make the process easier, we have provided a simple template of a GitHub Actions workflow to speed up the process of creating your own CI script/process. This template performs the following:
```

File: docs/direct_publish_setup.md [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+
+- Publishing to GitHub Releases, the Microsoft Marketplace and OpenVSX
+- Uploading the `.vsix` package to GitHub Releases as assets
+- Manual triggers that allow you to select what places to release to
```
Comment (loujaybee) [inline docs/direct_publish_setup.md:9]:
```suggestion
- Manual CI triggers for Open VSX and/or Microsoft Marketplace publishing.
```

File: docs/direct_publish_setup.md [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+- Automatic triggers from new GitHub Releases
+
+## Setup
+
+1. The first step is is to follow the [Publishing Extensions](https://github.com/eclipse/openvsx/wiki/Publishing-Extensions) doc and take note of the access token that you get as you will need it in the next step.
```
Comment (loujaybee) [inline docs/direct_publish_setup.md:14]:
```suggestion
1. The first step follows the [Publishing Extensions](https://github.com/eclipse/openvsx/wiki/Publishing-Extensions) doc and take note of the access token that you get as you will need it in the next step.
```

File: docs/direct_publish_setup.md [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+
+2. In order to run this the Action above, you need to setup two repository secrets for the Action to use:
```
Comment (loujaybee) [inline docs/direct_publish_setup.md:16]:
```suggestion
2. To run the GitHub Action above, you need to setup two repository secrets for the Action to use:
```

File: docs/extension_issues.md [hunk 20]
```diff
@@ -0,0 +1,29 @@
+# Bugs with extensions
+
+## Bugs in extensions' behaviors
+
+If an extension you downloaded from OpenVSX is misbehaving, please do the following:
```
Comment (loujaybee) [inline docs/extension_issues.md:5]:
```suggestion
If an extension you downloaded from OpenVSX isn't functioning as expected, please follow these steps to debug:
```

File: docs/extension_issues.md [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+
+1. check the extension version on [OpenVSX](https://open-vsx.org/) (you can see the latest one in the right sidebar)
+2. check the extension version on the [VS Code Marketplace](https://marketplace.visualstudio.com/) (you can find the version on the bottom of the right-hand side bar)
+3. If the versions don't match, it's probably a problem with the publishing process, so continue to the next chapter of this guide
+4. If they are the same version, please go to the problematic extension's repository and try to deal with the issue there.
```
Comment (loujaybee) [inline docs/extension_issues.md:10]:
```suggestion
4. If they are the same version, visit the extension's source repository to investigate the issue further.
```

File: docs/extension_issues.md [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+
+We cannot help with broken extensions; these issues have to be addressed with their respective maintainers.
```
Comment (loujaybee) [inline docs/extension_issues.md:12]:
```suggestion
**Important:** We (`publish-extensions` maintainers) cannot help with functionally broken extensions; these issues should be raised with the respective maintainers.
```

File: docs/extension_issues.md [hunk 23]
```diff
@@ -0,0 +1,29 @@
...
+
+If the extension at hand has been unmaintained for a longer period of time, please [raise an issue](https://github.com/open-vsx/publish-extensions/issues/new) by selecting <kbd>Other</kbd> as the issue template and adding the necessary information (we will publish the last version by building manually and remove the extension from this repository).
```
Comment (loujaybee) [inline docs/extension_issues.md:14]:
```suggestion
If the extension has been unmaintained for a longer period of time, please [raise an issue](https://github.com/open-vsx/publish-extensions/issues/new) by selecting <kbd>Other</kbd> as the issue template and adding the necessary information (we will publish the last version by building manually and remove the extension from this repository).
```

File: docs/direct_publish_setup.md (hunk 17 above)
Comment (filiptronicek) [reply docs/direct_publish_setup.md:9]: @loujaybee I think this new wording does not mention GitHub Releases, which are also a possibility. I am not sure how to rephrase it to still have the `and/or` part in it with three possible options 😄

File: docs/extension_issues.md [hunk 24]
```diff
@@ -0,0 +1,29 @@
...
+
+## Errors in the publishing process
+
+If an extension on OpenVSX is outdated or never successfully been published (although listed in [`extensions.json`](https://github.com/open-vsx/publish-extensions/blob/master/extensions.json)), this is probably due to one of two things:
```
Comment (loujaybee) [inline docs/extension_issues.md:18]:
```suggestion
If an extension on OpenVSX is outdated or has never been successfully published (despite being listed in [`extensions.json`](https://github.com/open-vsx/publish-extensions/blob/master/extensions.json)), the cause is likely one of the following:
```

File: docs/extension_issues.md [hunk 25]
```diff
@@ -0,0 +1,29 @@
...
+
+1. The extension has some abnormal build prerequisites, we build everything inside the same Ubuntu VM, which might cause problems like:
+    - a CLI tool is not installed
+    - the extension requires an older/newer version of Node.js: we are using Node 14 for everything.
+    - the extension has issues building on the latest LTS of Ubuntu server (we use `ubuntu-latest` for our jobs, you can take a look at [GitHub's Docs](https://github.com/actions/virtual-environments#available-environments) to see what that currently stands for)
+2. The extension requires additional commands to be executed to build successfully. 
+    - if you want a quick and easy fix you can try adding a `prepublish` property to the extension in `extensions.json` to set a command to be executed before packaging up the extension, right after installing the project's dependencies.
+
+The best way to solve this issue is to publish the extension from its own repository, not this one.
```
Comment (loujaybee) [inline docs/extension_issues.md:27]:
```suggestion
The best way to solve issues of failed publishing is to publish the extension from its own repository, not this one. 
```

File: docs/extension_issues.md [hunk 26]
```diff
@@ -0,0 +1,29 @@
...
+
+1. The extension has some abnormal build prerequisites, we build everything inside the same Ubuntu VM, which might cause problems like:
+    - a CLI tool is not installed
+    - the extension requires an older/newer version of Node.js: we are using Node 14 for everything.
+    - the extension has issues building on the latest LTS of Ubuntu server (we use `ubuntu-latest` for our jobs, you can take a look at [GitHub's Docs](https://github.com/actions/virtual-environments#available-environments) to see what that currently stands for)
+2. The extension requires additional commands to be executed to build successfully. 
+    - if you want a quick and easy fix you can try adding a `prepublish` property to the extension in `extensions.json` to set a command to be executed before packaging up the extension, right after installing the project's dependencies.
+
+The best way to solve this issue is to publish the extension from its own repository, not this one. 
+- If you are the extension author, you can use [this document](direct_publish_setup.md) for a guide on setting that up.
```
Comment (loujaybee) [inline docs/extension_issues.md:28]:
```suggestion
- If you are the extension author, you can use [this document](direct_publish_setup.md) as a guide on how to set up a CI publishing process.
```

File: docs/extension_issues.md [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+- If you are a community member you can raise an issue (if there is none discussing it already) using [this template](external_contribution_request.md). If the maintainers are willing to accept a contribution you can use the [same document listed in the point above](direct_publish_setup.md) to quickly setup a CI job with GitHub Actions.
```
Comment (loujaybee) [inline docs/extension_issues.md:29]:
```suggestion
- If you are a community member you can raise an issue (please check for existing issues) using [this template](external_contribution_request.md). If the maintainers are willing to accept a contribution you can use the [same document listed in the point above](direct_publish_setup.md) to quickly setup a CI job with GitHub Actions.
```

File: docs/external_contribution_request.md [hunk 28]
```diff
@@ -0,0 +1,17 @@
...
+
+The docs for publishing the extension are [here](https://github.com/eclipse/openvsx/wiki/Publishing-Extensions). Technically this should be straightforward. You get a token and call `ovsx publish` with it. For example, this is how RedHat publishes their Java extension: [redhat-developer/vscode-java@ff51dbf/Jenkinsfile#L73-L82](https://github.com/redhat-developer/vscode-java/blob/5d4f8d58b8e919534800ca7bc41e8513c288f573/Jenkinsfile#L78-L82)
+
+[Here](https://github.com/PeterWone/vsc-print/pull/121) is also an example PR that contributes a release GitHub Action which can publish to GitHub, MS and OpenVSX at the same time. Let @filiptronicek or @akosyakov know if you are open to a PR contributing such GitHub Action. They would gladly help out, even if there are just some questions or suggestions.
+
+Gitpod as well supports open-source developers by providing unlimited access to their platform. Feel free to reach out via https://www.gitpod.io/docs/professional-open-source
```
Comment (loujaybee) [inline docs/external_contribution_request.md:16]: I'm not so sure on the Gitpod plug in this template, as we don't want Gitpod to be mistaken as the direct owner of Open VSX. I also wouldn't necessarily mention Anton/yourself. 

```suggestion
Dear extension author,

Please publish this extension to the Open VSX marketplace. 

**Context**
Unfortunately, as Microsoft prohibits usages of Microsoft marketplace by any other products or redistribution vsix files from it, in order to use VS Code extensions in non-Microsoft products, we kindly ask that you take ownership of the VS Code extension namespace in [OpenVSX](https://open-vsx.org/) and publish this extension on Open VSX.

**What is OpenVSX? Why does it exist?**
OpenVSX is a vendor neutral alternative to MS marketplace used by all other derivatives of VS Code like [VSCodium](https://vscodium.com/), [Gitpod](gitpod.io), [OpenVSCode](https://github.com/gitpod-io/openvscode-server), [Theia](https://theia-ide.org/) based IDEs, and so on.

**How can you publish to Open VSX?**
The docs to publish an extension can be found [here](https://github.com/eclipse/openvsx/wiki/Publishing-Extensions). This process is straightforward and shouldn't take too long. Essentially, you need an authentication token and to execute the command `ovsx publish` to publish your plugin. See this example from RedHat publishes their Java extension: [redhat-developer/vscode-java@ff51dbf/Jenkinsfile#L73-L82](https://github.com/redhat-developer/vscode-java/blob/5d4f8d58b8e919534800ca7bc41e8513c288f573/Jenkinsfile#L78-L82)

You can also find [an example PR](https://github.com/PeterWone/vsc-print/pull/121) that contributes a release GitHub Action which publishes to GitHub, MS and OpenVSX at the same time. 
```

File: .github/ISSUE_TEMPLATES/bug.yml [hunk 29]
```diff
@@ -0,0 +1,23 @@
+name: Bug Report [publish-extensions]
+description: File a bug report about something wrong in the open-vsx/publish-extensions repository
```
Comment (loujaybee) [inline .github/ISSUE_TEMPLATES/bug.yml:2]:
```suggestion
description: A bug report about a failure to publish via the open-vsx/publish-extensions repository
```

File: .github/ISSUE_TEMPLATES/bug.yml [hunk 30]
```diff
@@ -0,0 +1,23 @@
...
+      value: |
+        Thanks for taking the time to fill out this bug report!
+  - type: textarea
+    id: describe
+    attributes:
+      label: Describe the bug
+      description: Please tell us what went wrong
+      value: "A bug happened!"
+    validations:
+      required: true
```
Comment (loujaybee) [inline .github/ISSUE_TEMPLATES/bug.yml:17]:
```suggestion
  - type: textarea
    id: describe
    attributes:
      label: Expected result and actual result
      description: What actually occurred, and what did you expect?
      value: "I expected X plugin to publish successfully, but it is currently intermittently failing"
    validations:
      required: true
```

File: .github/ISSUE_TEMPLATES/extension.yml [hunk 31]
```diff
@@ -0,0 +1,21 @@
...
+description: I want an extension to be published via this repo
+title: "[extension.id]: Publish to OpenVSX"
+labels: ["extension"]
+assignees: ["filiptronicek"]
+body:
+  - type: checkboxes
+    id: terms
+    attributes:
+      label: Pre-submission checklist
+      description: We require new publication requests to be first communicated with the extension's maintainers (see [why](https://github.com/open-vsx/publish-extensions/blob/master/README.md#when-to-add-an-extension)), only if they don't want to / can't publish it on their own or are unresponsive we will accept the request. If you haven't made an issue yet, you can create one in the extension's repository using [this template](https://github.com/open-vsx/publish-extensions/blob/master/docs/external_contribution_request.md).
```
Comment (loujaybee) [inline .github/ISSUE_TEMPLATES/extension.yml:11]:
```suggestion
      description: We require new publication requests to be first communicated with the extension's maintainers (see [why](https://github.com/open-vsx/publish-extensions/blob/master/README.md#when-to-add-an-extension)), it the author cannot publish the extension or is unresponsive, raise a PR. If you haven't made an issue for the extension owner yet, please create one in the extension's repository using [this template](https://github.com/open-vsx/publish-extensions/blob/master/docs/external_contribution_request.md).
```

File: docs/direct_publish_setup.md (hunk 17 above)
Comment (loujaybee) [reply docs/direct_publish_setup.md:9]: Updated! Does it work now? 😁

File: docs/direct_publish_setup.md [hunk 32]
```diff
@@ -0,0 +1,26 @@
...
+We advise extension authors to publish their extensions to OpenVSX as a part of their CI/CD process. See [our reasoning why that is](https://github.com/open-vsx/publish-extensions#when-to-add-an-extension).
+
+We have a simple template for setting a workflow with GitHub Actions which can help you kickstart your own. This template includes some neat features like:
+
+- Publishing to GitHub Releases, the Microsoft Marketplace and OpenVSX
+- Uploading the `.vsix` package to GitHub Releases as assets
+- Manual triggers that allow you to select what places to release to
+- Automatic triggers from new GitHub Releases
+
+## Setup
```
Comment (loujaybee) [inline docs/direct_publish_setup.md:12]:
```suggestion
## Setup VS Code extension publishing CI workflow
```

File: docs/extension_issues.md [hunk 33]
```diff
@@ -0,0 +1,29 @@
+# Bugs with extensions
+
+## Bugs in extensions' behaviors
```
Comment (loujaybee) [inline docs/extension_issues.md:3]:
```suggestion
## Functional bugs in VS Code extensions
```

File: docs/external_contribution_request.md [hunk 34]
```diff
@@ -0,0 +1,17 @@
+## Issue title
+
+Publish `EXTENSION_NAME` to Open VSX
```
Comment (loujaybee) [inline docs/external_contribution_request.md:3]: Should we just use this as the title, otherwise the copy/paster has to update/remove the placeholder title? 

```suggestion
## Publish `EXTENSION_NAME` to Open VSX

```

File: docs/external_contribution_request.md (hunk 8 above)
Comment (loujaybee) [reply docs/external_contribution_request.md:7]: I'd then remove all the boilerplate and simply keep the file to copy/paste the full thing? 🤔  

WDYT?

File: docs/direct_publish_setup.md (hunk 17 above)
Comment (filiptronicek) [reply docs/direct_publish_setup.md:9]: I think this will work, although it does not mention GitHub Releases explicitly, but it is stated two points above, so it isn't all that necessary.

File: docs/external_contribution_request.md (hunk 28 above)
Comment (filiptronicek) [reply docs/external_contribution_request.md:16]: This looks like a great change, @loujaybee! I would go so far as to add a section linking to the https://github.com/open-vsx/publish-extensions/blob/ft/issue-and-pr-templates/docs/direct_publish_setup.md doc and add a mention of us (or the person that creates the PR) being open to contributing the CI workflow to the target repository.

The idea with the mentions was by @akosyakov, so he probably has some thoughts 😃.

File: docs/external_contribution_request.md (hunk 34 above)
Comment (filiptronicek) [reply docs/external_contribution_request.md:3]: Hmm I think it wouldn't exactly make sense this way, because there is also a heading saying `Issue Body`. I think it will be even better if we make the issue title a code block the same way as the issue body.

We could also simply have the whole md file be the issue content (make the first heading imply to be used as the issue title), but in this way we can both add notes to the sections (maybe some guidance to the contributors), as well as providing a simple copying experience thanks to the code blocks.

File: docs/external_contribution_request.md (hunk 28 above)
Comment (akosyakov) [reply docs/external_contribution_request.md:16]: I'm fine to drop it. Put Gitpod there to motivate author a bit to become OpenVSX users through it.

Comment (brianking) [reply docs/external_contribution_request.md:16]: > I'm not so sure on the Gitpod plug in this template, as we don't want Gitpod to be mistaken as the direct owner of Open VSX. I also wouldn't necessarily mention Anton/yourself.

I agree. As a possible alternative, I am happy to collaborate on a resource that points people to ways they can try Open VSX. One idea is a page on open-vsx.org, a la https://github.com/EclipseFdn/open-vsx.org/issues/773

File: docs/external_contribution_request.md (hunk 34 above)
Comment (filiptronicek) [reply docs/external_contribution_request.md]: I implemented it as I said here, but we can change it anytime if it does not make sense to you 🙂.

**Comment by akosyakov** [comment]
good to merge?

**Comment by filiptronicek** [comment]
I think so!

File: docs/exampleCI.yaml [hunk 35]
```diff
@@ -0,0 +1,117 @@
...
+          release_name: Release ${{ needs.package.outputs.version }}
+          draft: false
+          prerelease: false
+
+      - name: Upload assets to a Release
+        uses: AButler/upload-release-assets@v2.0
+        with:
+          files: ${{ needs.package.outputs.packageName }}
+          release-tag: ${{ needs.package.outputs.tag }}
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Siddhant-K-code) [inline docs/exampleCI.yaml:117]: we can add a newline at EOF

File: .github/ISSUE_TEMPLATES/config.yml [hunk 36]
```diff
@@ -0,0 +1,5 @@
+blank_issues_enabled: false
+contact_links:
+  - name: Bug report [extension]
+    url: https://github.com/open-vsx/publish-extensions/blob/HEAD/docs/extension_issues.md
+    about: Please see this doc about reporting issues with extensions published from here
```
Comment (Siddhant-K-code) [inline .github/ISSUE_TEMPLATES/config.yml:5]: we can add a newline at EOF

**Review by Siddhant-K-code — COMMENTED** [review]
adding a newline at EOF

File: docs/exampleCI.yaml (hunk 35 above)
Comment (filiptronicek) [reply docs/exampleCI.yaml:117]: Thank you!
